### PR TITLE
add shebang support

### LINKED ISSUE #251: Add shebang support (#!)
To use tengo as scripting language in the terminal, support for shebang (`#!`) would be beneficial.

Additional question, what is the proposed file extension for tengo files?

### PR DESCRIPTION
**Pull request by ozanh**
This PR should add a simple Shebang support for Tengo scripts and closes #251

### CODE REVIEW

**Comment by ozanh** [comment]
Should I add skipShebang function to compile only mode?

**Comment by d5** [comment]
Could be wrong, but I think shebang must be at the beginning of the script (file). I think most tools will assume the same. So I think the code can be simpler?

**Comment by ozanh** [comment]
I checked https://en.wikipedia.org/wiki/Shebang_(Unix) and before `#!` spaces are allowed. I can remove `\r` but spaces and tabs are allowed. For example following works at my Linux machine;

```bash
# bash/sh allow spaces before and after #!
echo -e '\t #! /bin/bash \t\necho works' > test
chmod +x test
./test # works
```

```bash
# python2/3 only allow spaces after #! not before
echo -e '#! \t/usr/bin/python3 \t\nprint("works")' > test.py
chmod +x test.py
./test.py # works
```
Anything after executable's path must also be allowed because command line arguments can be passed to executable as well. I should remove `\r` but not sure about the rest.
How should I proceed?

**Comment by geseq** [comment]
I like the way yaegi does it

**Comment by ozanh** [comment]
I think yaegi copies whole buffer to a new string which may not be necessary.

File: cmd/tengo/main.go [hunk 1]
```diff
@@ -67,6 +67,10 @@ func main() {
 			os.Exit(1)
 		}
 	} else if filepath.Ext(inputFile) == sourceFileExt {
+		if len(inputData) > 1 &&
+			bytes.Compare(inputData[:2], []byte("#!")) == 0 {
+			copy(inputData, "//")
```
Comment (d5) [inline cmd/tengo/main.go:72]: 🤣 this is smart!!!

**Comment by d5** [comment]
Looks good to me!

**Comment by d5** [comment]
@geseq feel free to review and merge

**Pull request merged**